composite_broker/src/broker.rs

```diff
@@ -0,0 +1,298 @@
+pub mod broker {
+    use rand::prelude::*;
+    use std::{collections::HashMap};
+    // use mqtt_v5::{topic::TopicFilter, types::{ConnectAckPacket, ConnectPacket, ConnectReason, 
+    //     properties::AssignedClientIdentifier, SubscribeAckPacket, SubscribePacket, SubscribeAckReason,
+    //     properties::ReasonString, PublishPacket, PublishAckPacket, PublishAckReason, Packet, self
+    // }};
+    use mqtt_sn::{self, Message, Connect, ConnAck, ReturnCode, RejectedReason, Subscribe, SubAck, 
+        Flags, Publish, PubAck, UnsubAck, Unsubscribe}; // TopicName
+
+    
+    #[derive(Debug, Clone, PartialEq)]
+    #[allow(dead_code)]
+    pub struct Client {
+        client_id: String,
+        address: String,
```

Comment:
I'd imagine these would better be stored as numbers, not as strings. String comparisons are expensive, in general, so you only want to use them if you have to. Just a note.

Reply:
Would storing them as [type IPv4](https://doc.rust-lang.org/std/net/struct.Ipv4Addr.html) be just as expensive?